### PR TITLE
fix(PipelineDialog): prevent display of empty value on picker

### PR DESCRIPTION
**Pull request by cheikhgwane**
Filter and display the correct connection from pipeline run config.

A short, meaningful PR description. Explain the goal of the PR and the ideas behind it.

## Changes

Please list / describe the changes in the codebase for the reviewer(s).

- Filter and display connection based on the value (if empty return null)
- Change WorkspaceConnectionPicker value type (from Option to string)
- Pass the connection slug to onChange method in ParameterField instead of connection object

## Screenshots / screencast

[Screencast from 11-22-2023 12:53:39 PM.webm](https://github.com/BLSQ/openhexa-frontend/assets/25453621/47053bc8-4288-4502-99c2-f92d7228f0ef)

### CODE REVIEW

File: src/workspaces/features/WorkspaceConnectionPicker/WorkspaceConnectionPicker.tsx [hunk 1]
```diff
@@ -74,20 +74,24 @@ const WorkspaceConnectionPicker = (props: WorkspaceConnectionPickerProps) => {
     [],
   );
 
+  const defaultValue = useMemo(() => {
```
Comment (pvanliefland) [inline src/workspaces/features/WorkspaceConnectionPicker/WorkspaceConnectionPicker.tsx:77]: Let's call it `comboBoxValue`.

**Review by pvanliefland — CHANGES_REQUESTED** [review]
As discussed

**Review by pvanliefland — APPROVED** [review]
LGTM!

File: src/workspaces/features/WorkspaceConnectionPicker/WorkspaceConnectionPicker.tsx [hunk 2]
```diff
@@ -74,20 +74,24 @@ const WorkspaceConnectionPicker = (props: WorkspaceConnectionPickerProps) => {
...
+
   return (
     <Combobox
       required={required}
       onChange={onChange}
       loading={loading}
       withPortal={withPortal}
       displayValue={displayValue}
-      by={(a: Option, b: Option) => a.id === b.id}
+      by="name"
```
Comment (qgerome) [inline src/workspaces/features/WorkspaceConnectionPicker/WorkspaceConnectionPicker.tsx:88]: Looks weird to me that we use the "name" of the connection as the identifier of the connection.

**Pull request merged**